common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -550,24 +550,36 @@ static void putLong(byte[] data, int index, long value) {
     }
 
     static void copyMemory(long srcAddr, long dstAddr, long length) {
-        //UNSAFE.copyMemory(srcAddr, dstAddr, length);
-        while (length > 0) {
-            long size = Math.min(length, UNSAFE_COPY_THRESHOLD);
-            UNSAFE.copyMemory(srcAddr, dstAddr, size);
-            length -= size;
-            srcAddr += size;
-            dstAddr += size;
+        // Manual safe-point polling is only needed prior Java9:
+        // See https://bugs.openjdk.java.net/browse/JDK-8149596
+        if (javaVersion() <= 8) {
+            //UNSAFE.copyMemory(srcAddr, dstAddr, length);
+            while (length > 0) {
+                long size = Math.min(length, UNSAFE_COPY_THRESHOLD);
+                UNSAFE.copyMemory(srcAddr, dstAddr, size);
+                length -= size;
+                srcAddr += size;
+                dstAddr += size;
+            }
+        } else {
+            UNSAFE.copyMemory(srcAddr, dstAddr, length);
         }
     }
 
     static void copyMemory(Object src, long srcOffset, Object dst, long dstOffset, long length) {
-        //UNSAFE.copyMemory(src, srcOffset, dst, dstOffset, length);
-        while (length > 0) {
-            long size = Math.min(length, UNSAFE_COPY_THRESHOLD);
-            UNSAFE.copyMemory(src, srcOffset, dst, dstOffset, size);
-            length -= size;
-            srcOffset += size;
-            dstOffset += size;
+        // Manual safe-point polling is only needed prior Java9:
+        // See https://bugs.openjdk.java.net/browse/JDK-8149596
+        if (javaVersion() <= 8) {
+            //UNSAFE.copyMemory(src, srcOffset, dst, dstOffset, length);
```

Comment:
Not a part of this commit, but is this commented out code necessary? Same above.